src/decoders/brainfuck_interpreter.rs

```diff
@@ -0,0 +1,187 @@
+//! Run a brainfuck program and return its output
+//! Performs error handling and returns a string
+//! Call brainfuck_interpreter.crack to use. It returns Option<String> and check with
+//! `result.is_some()` to see if it returned okay.
+use crate::checkers::CheckerTypes;
+
+use super::crack_results::CrackResult;
+use super::interface::Crack;
+use super::interface::Decoder;
+
+use brainfuck_exe::Brainfuck;
+use log::{debug, trace};
+
+/// The Brainfuck interpreter, call:
+/// `let brainfuck_interpreter = Decoder::<BrainfuckInterpreter>::new()` to create a new instance
+/// And then call:
+/// `result = brainfuck_interpreter.crack(input)` to interpret a Brainfuck program
+/// The struct generated by new() comes from interface.rs
+/// ```
+/// use ares::decoders::brainfuck_interpreter::{BrainfuckInterpreter};
+/// use ares::decoders::interface::{Crack, Decoder};
+/// use ares::checkers::{athena::Athena, CheckerTypes, checker_type::{Check, Checker}};
+///
+/// let brainfuck_interpreter = Decoder::<BrainfuckInterpreter>::new();
+/// let athena_checker = Checker::<Athena>::new();
+/// let checker = CheckerTypes::CheckAthena(athena_checker);
+///
+/// let result = brainfuck_interpreter.crack(">++++++++[<+++++++++>-]<.>++++[<+++++++>-]<+.+++++++..+++.>>++++++[<+++++++>-]<++.------------.>++++++[<+++++++++>-]<+.<.+++.------.--------.>>>++++[<++++++++>-]<+.", &checker).unencrypted_text;
+/// assert!(result.is_some());
+/// assert_eq!(result.unwrap()[0], "Hello, World!");
+/// ```
+pub struct BrainfuckInterpreter;
+
+impl Crack for Decoder<BrainfuckInterpreter> {
+    fn new() -> Decoder<BrainfuckInterpreter> {
+        Decoder {
```

Comment:
Todo list for me, in A* heuristics when it detects say Vigenere it prioritises that using cipher identifier

I think adding brainfuck to the regex would be good...